src/unit-test-coverage/shared/src/coveragetest-idmap.c

```diff
@@ -1065,6 +1065,76 @@ void Test_OS_ObjectIdIterator(void)
     UtAssert_STUB_COUNT(OS_Unlock_Global_Impl, 2);
 }
 
+void Test_OS_ObjectIDInteger(void)
+{
+    /*
+     * Test Case For:
+     * OS_ObjectIdToInteger, OS_ObjectIdFromInteger, OS_ObjectIdEqual, OS_ObjectIdDefined
+     */
+    int32             actual;
+    OS_object_token_t token;
+    osal_id_t         typesI[OS_MAX_TOTAL_RECORDS];
+    osal_id_t         typesJ[OS_MAX_TOTAL_RECORDS];
+    uint32            intID;
+    int32             recordscount = 0;
+    osal_objtype_t    idtype;
+    char              str[OS_MAX_API_NAME];
+
+    for (idtype = 0; idtype < OS_OBJECT_TYPE_USER; ++idtype)
+    {
+        actual = OS_SUCCESS;
+        while (actual == OS_SUCCESS && recordscount < OS_MAX_TOTAL_RECORDS)
+        {
+            snprintf(str, sizeof(str), "%d", (int)recordscount);
+            actual = OS_ObjectIdAllocateNew(idtype, str, &token);
+
+            if (actual == OS_SUCCESS)
+            {
+                typesI[recordscount] = token.obj_id;
+                intID                = OS_ObjectIdToInteger(typesI[recordscount]);
+                typesJ[recordscount] = OS_ObjectIdFromInteger(intID);
+
+                recordscount++;
+            }
+        }
+    }
+
+    UtAssert_True(recordscount < OS_MAX_TOTAL_RECORDS, "All Id types checked");
+
+    for (int i = 0; i < recordscount; i++)
+    {
+        UtAssert_True(OS_ObjectIdDefined(typesI[i]), "%lu Is defined", OS_ObjectIdToInteger(typesI[i]));
+
+        for (int j = 0; j < recordscount; j++)
+        {
+            if (i == j)
+            {
+                UtAssert_True(OS_ObjectIdEqual(typesI[i], typesJ[j]), "%lu equals %lu", OS_ObjectIdToInteger(typesI[i]),
+                              OS_ObjectIdToInteger(typesJ[j]));
+            }
+            else
+            {
+                UtAssert_True(!OS_ObjectIdEqual(typesI[i], typesJ[j]), "%lu does not equal %lu",
```

Comment:
This line is what produces the majority of the 79243 cases and makes the log huge.